jbehave-support-core/src/main/java/org/jbehavesupport/core/internal/web/WebScreenshotCreator.java

```diff
@@ -61,23 +85,30 @@ private void prepareDirectory() throws IOException {
 
     private File getDestinationFile() {
         if (testContext.contains(AbstractSpringStories.JBEHAVE_SCENARIO)) {
-            final String storyName = testContext.get(AbstractSpringStories.JBEHAVE_SCENARIO, String.class).split("#")[0];
-            File destinationFile = new File(screenshotDirectory, String.format(FILE_NAME_PATTERN, storyName));
+            String storyName = testContext.get(AbstractSpringStories.JBEHAVE_SCENARIO, String.class).split("#")[0];
+            File destinationFile = new File(screenshotDirectory, String.format(FILE_NAME_PATTERN, screenShotType, storyName));
             int i = 1;
             while (destinationFile.exists()) {
-                destinationFile = new File(screenshotDirectory, String.format(FILE_NAME_PATTERN, storyName + "-" + i++));
+                destinationFile = new File(screenshotDirectory, String.format(FILE_NAME_PATTERN, screenShotType, storyName + "-" + i++));
             }
             return destinationFile;
         } else {
             return new File("screenshot-" + System.currentTimeMillis() + ".png");
         }
     }
 
-    private void storeInTestContext(final String destFile) {
-        if (!testContext.contains(SCREENSHOTS_KEY)) {
-            testContext.put(SCREENSHOTS_KEY, new LinkedHashSet<String>());
+    private void storeFailedInTestContext(final String destFile) {
```

Comment:
Can probably be merge into one `storeInTestContext(String destFile, String screenshotsKey)`, e.g. `storeInTestContext(destinationFile.getName(), STEP_SCREENSHOTS_KEY)`

Reply:
Merged, I will remove unused methods in final version.